src/SizedArray.jl

```diff
@@ -53,43 +64,156 @@ end
         return a
     end
 end
-
-@inline SizedArray{S,T,N}(x::Tuple) where {S,T,N} = SizedArray{S,T,N,N}(x)
-@inline SizedArray{S,T}(x::Tuple) where {S,T} = SizedArray{S,T,tuple_length(S),tuple_length(S)}(x)
-@inline SizedArray{S}(x::NTuple{L,T}) where {S,T,L} = SizedArray{S,T,tuple_length(S),tuple_length(S)}(x)
+@inline function SizedArray{S,T,N,M}(x::Tuple) where {S,T,N,M}
+    return SizedArray{S,T,N,M,Array{T,M}}(x)
+end
+@inline function SizedArray{S,T,N}(x::Tuple) where {S,T,N}
+    return SizedArray{S,T,N,N,Array{T,N}}(x)
+end
+@inline function SizedArray{S,T}(x::Tuple) where {S,T}
+    return SizedArray{S,T,tuple_length(S)}(x)
+end
+@inline function SizedArray{S}(x::NTuple{L,T}) where {S,T,L}
+    return SizedArray{S,T}(x)
+end
 
 # Overide some problematic default behaviour
 @inline convert(::Type{SA}, sa::SizedArray) where {SA<:SizedArray} = SA(sa.data)
 @inline convert(::Type{SA}, sa::SA) where {SA<:SizedArray} = sa
 
 # Back to Array (unfortunately need both convert and construct to overide other methods)
-@inline Array(sa::SizedArray) = Array(sa.data)
-@inline Array{T}(sa::SizedArray{S,T}) where {T,S} = Array{T}(sa.data)
-@inline Array{T,N}(sa::SizedArray{S,T,N}) where {T,S,N} = Array{T,N}(sa.data)
+@inline function Base.Array(sa::SizedArray{S}) where {S}
+    return Array(reshape(sa.data, size_to_tuple(S)))
+end
+@inline function Base.Array{T}(sa::SizedArray{S,T}) where {T,S}
+    return Array(reshape(sa.data, size_to_tuple(S)))
+end
+@inline function Base.Array{T,N}(sa::SizedArray{S,T,N}) where {T,S,N}
+    return Array(reshape(sa.data, size_to_tuple(S)))
+end
 
-@inline convert(::Type{Array}, sa::SizedArray) = sa.data
-@inline convert(::Type{Array{T}}, sa::SizedArray{S,T}) where {T,S} = sa.data
-@inline convert(::Type{Array{T,N}}, sa::SizedArray{S,T,N}) where {T,S,N} = sa.data
+@inline function convert(::Type{Array}, sa::SizedArray{S}) where {S}
```

Comment:
This only valid if `sa.data isa Array` - so we'll need a fallback that converts if not.

Reply:
Oh never mind I see you did it below!